nmostesting/ControllerTest.py

```diff
@@ -217,7 +219,11 @@ def _send_testing_facade_questions(
 
     def _wait_for_testing_facade(self, question_id, test_type, timeout=None):
 
-        question_timeout = (timeout or self.question_timeout) + self.extra_time
+        question_timeout = timeout or CONFIG.CONTROLLER_TESTING_TIMEOUT
+
+        # If timeout == 0 then get_answer_response will block until complete
+        # Add API processing time to specified timeout
+        question_timeout = question_timeout + (2 * CONFIG.API_PROCESSING_TIMEOUT) if question_timeout > 0 else None
```

Comment:
The comment is confusing because it's not clear if it's talking about the `timeout` argument to `get_answer_response` or the `timeout` parameter of this function. According to the docs for `[asyncio.wait_for](https://docs.python.org/3/library/asyncio-task.html#asyncio.wait_for)`, it will block given `None`. It doesn't explicitly mention `0`, so I'd expect that to check and return immediately.

Reply:
OK will improve comment.

Yes, you're right, which is why the 0 is converted to a `None` before being passed to `asyncio.wait_for`.